src/Sentry/Protocol/Context/Device.cs

```diff
@@ -222,5 +191,202 @@ internal Device Clone()
+            {
+                writer.WriteString("arch", Architecture);
+            }
+
+            if (BatteryLevel is {} batteryLevel)
+            {
+                writer.WriteNumber("battery_level", batteryLevel);
+            }
+
+            if (IsCharging is {} isCharging)
+            {
+                writer.WriteBoolean("charging", isCharging);
+            }
+
+            if (IsOnline is {} isOnline)
+            {
+                writer.WriteBoolean("online", isOnline);
+            }
+
+            if (Orientation is {} orientation)
+            {
+                writer.WriteString("orientation", orientation.ToString().ToLowerInvariant());
+            }
+
+            if (Simulator is {} simulator)
+            {
+                writer.WriteBoolean("simulator", simulator);
+            }
+
+            if (MemorySize is {} memorySize)
+            {
+                writer.WriteNumber("memory_size", memorySize);
+            }
+
+            if (FreeMemory is {} freeMemory)
+            {
+                writer.WriteNumber("free_memory", freeMemory);
+            }
+
+            if (UsableMemory is {} usableMemory)
+            {
+                writer.WriteNumber("usable_memory", usableMemory);
+            }
+
+            if (LowMemory is {} lowMemory)
+            {
+                writer.WriteBoolean("low_memory", lowMemory);
+            }
+
+            if (StorageSize is {} storageSize)
+            {
+                writer.WriteNumber("storage_size", storageSize);
+            }
+
+            if (FreeStorage is {} freeStorage)
+            {
+                writer.WriteNumber("free_storage", freeStorage);
+            }
+
+            if (ExternalStorageSize is {} externalStorageSize)
+            {
+                writer.WriteNumber("external_storage_size", externalStorageSize);
+            }
+
+            if (ExternalFreeStorage is {} externalFreeStorage)
+            {
+                writer.WriteNumber("external_free_storage", externalFreeStorage);
+            }
+
+            if (!string.IsNullOrWhiteSpace(ScreenResolution))
+            {
+                writer.WriteString("screen_resolution", ScreenResolution);
+            }
+
+            if (ScreenDensity is {} screenDensity)
+            {
+                writer.WriteNumber("screen_density", screenDensity);
+            }
+
+            if (ScreenDpi is {} screenDpi)
+            {
+                writer.WriteNumber("screen_dpi", screenDpi);
+            }
+
+            if (BootTime is {} bootTime)
+            {
+                writer.WriteString("boot_time", bootTime);
+            }
+
+            writer.WriteEndObject();
+        }
+
+        /// <summary>
+        /// Parses from JSON.
+        /// </summary>
+        public static Device FromJson(JsonElement json)
+        {
+            var timezone = json.GetPropertyOrNull("timezone")?.GetString()?.Pipe(TimeZoneInfo.FindSystemTimeZoneById);
```

Comment:
We really need a test for this given that tz information on linux is very different than windows. Even though we don't have a requirement now to get an envelope serialized on windows to be deserialized on linux, I just want to make sure it round trips on each OS independently

Reply:
There are tests that call `FromJson` on `Device` already.